site/app/templates/submission/homework/SubmitBox.twig

```diff
@@ -149,14 +149,74 @@
         <samp>git  clone  {{ repository_path }}  SPECIFY_TARGET_DIRECTORY</samp><br /><br />
         <input type="submit" id="submit" class="btn btn-primary" value="Grade My Repository" />
     {% else %}
-        <div id="upload-boxes" style="display:table; border-spacing: 5px; width:100%">
+      <div id="upload-boxes" style="display:table; border-spacing: 5px; width:100%">
 
-            {# Text boxes #}
-            {% for textbox in textboxes %}
-                {{ self.render_textbox(textbox, image_data, loop.index0) }}
-            {% endfor %}
-            {# /Text boxes #}
+	<script type="text/javascript" src="../../../../js/showdown.min.js"></script>
+	<script type="text/javascript" src="../../../../js/xss.min.js"></script>
+	<script type="text/javascript" src="../../../../js/showdown-xss-filter.js"></script>
+
+	{# Render content #}
+	{% for c in contents %}
+	  <h2>{{ c.title }}</h2>
+	  <div id="exp_{{ loop.index0 }}"></div>
+	  <script type="text/javascript">
+	    var converter = new showdown.Converter({ extensions: ['xssfilter']});
+	    var md = "{{ c.exposition|raw }}";
+	    document.getElementById("exp_{{ loop.index0 }}").innerHTML = converter.makeHtml(md);
+	  </script>
+	  <br>
+
+	    {# Render the inputs for the given content block #}
+	    {% for i in c.input %}
+	      {% if i.type == "textbox" %}
+		{{ self.render_textbox(i, image_data, loop.index0) }}
+	      {% elseif i.type == "codebox" %}
+		<style type="text/css" media="screen">
+		  #editor_{{ loop.index0 }} { 
+		  width: 75%;
+		  border: 3px solid #FF00FF;
+		  }
+		</style>
```

Comment:
Change this to a class and put it into `server.css`. Then give all divs that class.